rpaas/manager.py

```diff
@@ -251,11 +256,20 @@ def update_certificate(self, name, cert, key):
         self.storage.update_binding_certificate(name, cert, key)
         self.consul_manager.set_certificate(name, cert, key)
 
-    def add_upstream(self, name, upstream_name, server):
+    def add_upstream(self, name, upstream_name, server, acl=False):
         self.task_manager.ensure_ready(name)
         lb = LoadBalancer.find(name)
         if lb is None:
             raise storage.InstanceNotFoundError()
+        if acl:
+            for host in lb.hosts:
+                if isinstance(server, list):
```

Comment:
What about changing this to 
```
if not isinstance(server, list):
    server = [server]
```
And then having a single code path instead of an if/else